### ISSUE TITLE
Missing counter + timestamp

### ISSUE THREAD

**Issue by ericelliott** [overflowdigital/cuid2.py#2]
This implementation seems to be placing too much faith in the system random number generator. We consider this to be unsafe because many system number generators ("CSRNGs" or otherwise) have had critical bugs which allowed attackers to create predict upcoming random numbers. Since the other values going into the id are fixed (Pid, hostname), an attacker who knows those properties could predict Cuids generated by this implementation.

Conformant implementations of Cuid2 will use a millisecond-resolution timer and a randomly-initialized auto-incrementing counter.

**Comment by joshuathompsonlindley** [comment]
Thanks for the issue. We've updated the library (https://github.com/overflowdigital/cuid2.py/actions/runs/3882764236) to use a ms timestamp and to use a wrapper around urandom/CryptGenRandom.

**Comment by ericelliott** [comment]
I looked at the code and did not see a counter.

On many systems, there is not sufficient entropy within the same ms to reliably produce different random number outputs. When those systems generate ids in a tight loop, they can produce collisions due to lack of entropy. This is why we added the session counter in Cuid v1, which inspired the [Monotonicity and Counters](https://www.ietf.org/archive/id/draft-peabody-dispatch-new-uuid-format-04.html#name-monotonicity-and-counters) portion of the new UUID v6-v8 standards. The spec describes some conditional logic to only increment a counter if the id is generated in the same ms (or whatever resolution your timer uses). Cuid (v1 and v2) just always increment a session counter and mix it in with the rest of the id entropy.
